beacon_node/network/src/sync/backfill_sync/mod.rs

```diff
@@ -372,7 +372,9 @@ impl<T: BeaconChainTypes> BackFillSync<T> {
             // A batch could be retried without the peer failing the request (disconnecting/
             // sending an error /timeout) if the peer is removed from the chain for other
             // reasons. Check that this block belongs to the expected peer
-            if !batch.is_expecting_block(peer_id, &request_id) {
+            // TODO(das): removed peer_id matching as the node may request a different peer for data
+            // columns.
```

Comment:
this may be a permanent change now - but i guess we can keep this until the range sync rework happens to be sure.